### ISSUE TITLE
xcode_backend.sh No such file or directory

### ISSUE THREAD

**Issue by artAst** [flutter/flutter#17234]
I got the following error: `/bin/sh: /packages/flutter_tools/bin/xcode_backend.sh: No such file or directory`

I already did `flutter run` as indicated here: https://github.com/flutter/flutter/issues/8011

but still the same issue occurs.

## Flutter Doctor

[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-US, channel alpha)
    • Flutter at /Applications/Flutter
    • Framework revision fd7853faad (5 months ago), 2017-12-05 16:12:55 -0800
    • Engine revision b57fca02b5
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d4cfecb1065d322d3670df7e9ec9a0cc2d4b90f0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /usr/local/Caskroom/android-sdk/3859397,26.0.2
    • Android NDK location not configured (this optional configuration is useful                                                                               for native profiling support)
    • Platform android-26, build-tools 26.0.2
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk1.8.0_152.jdk/Content                                                                              s/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_152-b16)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.3, Build version 9E145
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.0

[✗] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/stud                                                                              io/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instruction                                                                              s).

[✓] Connected devices
    • None

**Comment by ko514615** [comment]
I did run into the same error yesterday. 
Did you close the issue because you found a resolution?

If so, could you share your resolution?

Thanks

**Comment by st-f** [comment]
I'm having the same issue when running some of the sample projects (stocks and flutter_view), and flutter doctor returns no issue.

**Comment by hidrodixtion** [comment]
To manually fix this, open Project Target and add User-Defined Setting with value **FLUTTER_ROOT** with value  of your flutter sdk location.

![](https://i.imgur.com/v7WIYH2.png)

**Comment by lance-blo** [comment]
Right, those global variables had incorrect paths in my project. They had Windows paths (project imported from Win10) in an iOS project.

**Comment by artAst** [comment]
@hidrodixtion is correct. you need to add flutter_root.
This problem occurred on my dev server. It is not an issue from my local.
My dev setup was different, probably due to user permissions

**Comment by zoechi** [comment]
@lance-blo using Git to share projects with other machines helps prevent such issues because the 3 included `.gitignore` files prevent generated files from being committed.

**Comment by lance-blo** [comment]
Right, I just set it up this morning. Much cleaner like this.

**Comment by appleboy** [comment]
It is working with add `FLUTTER_ROOT` global variable in bash file `.bashrc`.

**Comment by kevinkicho** [comment]
User Defined Setting can be added by going to Editor --> Add Build Setting --> Add User-Defined Setting

**Comment by kangwang1988** [comment]
It's supposed to be managed by Generated.xcconfig(create/update by flutter_tools). I don't think write it using hard code like .bashrc or User-Defined Setting is a good idea.
If this issue can be reproduced even if you've run `flutter run` successfully, It should be a bug.

**Comment by AllenDongCoder** [comment]
what's the value FLUTTER_ROOT ,why i still can't run my project

**Comment by liaoyuanng** [comment]
run `flutter run` in `flutter_moudle`

**Comment by l-k22** [comment]
> `FLUTTER_ROOT` is supposed to be managed by `Generated.xcconfig`(create/update by flutter_tools). I don't think write it using hard code like `.bashrc` or User-Defined Setting is a good idea.
> If this issue can be reproduced even if you've run `flutter run` successfully, It should be a bug.

Did you ever come across a solution?

**Comment by SebastianEngel** [comment]
None of the Flutter variables defined in Generated.xcconfig should be added to the user defined build settings manually. 

When `flutter pub get`is run, it generates a file called `Generated.xcconfig` with all of the Flutter variables defined with values for the machine it is running on.

The build configuration files `Debug.xcconfig` and `Release.xcconfig` (in the `Flutter` folder) include `Generated.xcconfig`. Now you only have to ensure that for each or your project configuration these build configuration files are set properly (in XCode unter _Project > Info > Configuration_).

When you then go to _'Build Settings'_ and scroll down to _'User Defined'_, you should see the Flutter variables with the right values.

**Comment by timxschroeder** [comment]
@SebastianEngel is the only one correctly addressing the issue. Manually modifying these values (as I did) will give you a headache later on if you try to build any kind of CD pipeline where these paths are obviously going to be different. Really make sure to add the generated config to your build configuration for release and debug as he describes.

**Comment by pawlowskim** [comment]
I will leave this just for reference. I have also had this problem, when I was manipulating config files. I didn't add `#include "Flutter/Generated.xcconfig"` to my configs. So all the flutter vars weren't visible for xcode builder.

**Comment by gtupak** [comment]
I had the same issue and solved it by doing what @pawlowskim and @SebastianEngel said, plus I removed the values for the `FLUTTER_ROOT` variable.

**Comment by roberto198907** [comment]
I was having the same issue, the problem was I was opening Runner.xcodeproj intead of Runner.xcworksace, maybe someone of you could be making the same mistake.

**Comment by subjectdenied** [comment]
so setting FLUTTER_ROOT as an env is enough? cause obviouly "flutter clean" under linux also rewrites the Generated.xcconfig and puts the wrong flutter path there - in my case when importing a project from linux to mac it took over the wrong path (with FLUTTER_ROOT set as an env in linux, while not set on mac)

**Comment by sgehrman** [comment]
still having this problem.  nothing helped.

**Comment by dawidb-appdate** [comment]
@sgehrman make sure you User-Defined section is not changed manually. In my case it was manually edited (indication - bolded letters), so to fix it:
- remove all the bolded  entries in User-Defined section (Target->Build Settings)
- in Project settings->Info->Configuration section: open Scheme you're building and change the configuration to something else and then select again the previous one. 
It should re-generate all the entries in User-Defined section including Generated.xcconfig which contains all the valid paths
Make sure to have it under git control so you can easily go back to original code if something goes wrong.
<img width="1639" alt="Screenshot 2020-04-16 at 17 53 01" src="https://user-images.githubusercontent.com/2346413/79478828-2c55ba00-800c-11ea-9200-0e3473e278e9.png">

**Comment by shinriyo** [comment]
`FLUTTER_ROOT` can put relative path?

**Comment by YLoveX** [comment]
I use **App Center** to auto build flutter project, and i met this error too.
My **User-Defined** isn't **bold** and i still got error from **App Center**.
Local is fine.

Error is:
```
/bin/sh: /packages/flutter_tools/bin/xcode_backend.sh: No such file or directory
```

Anyone who can help me?

**Comment by saturov** [comment]
> @sgehrman make sure you User-Defined section is not changed manually. In my case it was manually edited (indication - bolded letters), so to fix it:
> 
> * remove all the bolded  entries in User-Defined section (Target->Build Settings)
> * in Project settings->Info->Configuration section: open Scheme you're building and change the configuration to something else and then select again the previous one.
>   It should re-generate all the entries in User-Defined section including Generated.xcconfig which contains all the valid paths
>   Make sure to have it under git control so you can easily go back to original code if something goes wrong.
> 
> <img alt="Screenshot 2020-04-16 at 17 53 01" width="1639" src="https://user-images.githubusercontent.com/2346413/79478828-2c55ba00-800c-11ea-9200-0e3473e278e9.png">

Thank you! That helps me a lot.

**Comment by dorbor** [comment]
> To manually fix this, open Project Target and add User-Defined Setting with key name **FLUTTER_ROOT** with the value of your flutter SDK location.
> 
> ![](https://camo.githubusercontent.com/3d63f9a8e94c795d3c16425f4000bbb15e260225/68747470733a2f2f692e696d6775722e636f6d2f763757495948322e706e67)

U JUST MADE MY DAY

**Comment by markathomas** [comment]
So I have Debug and Release build configs but also Pods-Runner.{debug,release,profile} which it was set to so changing the debug and release to Debug and Release, respectively, solved the issue.

**Comment by maksandaruwan** [comment]
in my case It was null those values highlighted in below image, so I selected relevant values. Build succeeded.
![Screen Shot 2020-11-14 at 12 49 16 PM](https://user-images.githubusercontent.com/18378260/99143621-622d5080-2678-11eb-994f-d129ee84e4f1.png)

**Comment by jhubbardsf** [comment]
@maksandaruwan Shouldn't profile be Generated or am I wrong there?

**Comment by Aaron009** [comment]
> To manually fix this, open Project Target and add User-Defined Setting with key name **FLUTTER_ROOT** with the value of your flutter SDK location.
> 
> ![](https://camo.githubusercontent.com/6168b8e48f348de139dedc03da9ae1510d42904d1fdbb2a2b8e01e424c9a9fff/68747470733a2f2f692e696d6775722e636f6d2f763757495948322e706e67)

Please update it and add this sentence. Many people don’t know where to add.

User Defined Setting can be added by going to Editor --> Add Build Setting --> Add User-Defined Setting
